hazelcast/src/test/java/com/hazelcast/spi/impl/operationservice/impl/InvocationNestedLocalTest.java

```diff
@@ -8,115 +8,120 @@
-    public void whenPartition_callsCorrectPartition() {
-        HazelcastInstance hz = createHazelcastInstance();
-        OperationService operationService = getOperationService(hz);
+    public void invokeOnPartition_outerLocal_innerSameInstance_samePartition() {
+        HazelcastInstance local = createHazelcastInstance();
+        OperationService operationService = getOperationService(local);
 
-        int partitionId = 0;
-        InnerOperation innerOperation = new InnerOperation(response, partitionId);
+        int partitionId = getPartitionId(local);
+        InnerOperation innerOperation = new InnerOperation(RESPONSE, partitionId);
         OuterOperation outerOperation = new OuterOperation(innerOperation, partitionId);
-
         InternalCompletableFuture f = operationService.invokeOnPartition(null, outerOperation, outerOperation.getPartitionId());
 
-        assertEquals(response, f.getSafely());
+        assertEquals(RESPONSE, f.getSafely());
     }
 
     @Test
-    public void whenPartition_callsIncorrectPartition() {
-        HazelcastInstance hz = createHazelcastInstance();
-        OperationService operationService = getOperationService(hz);
-
-        int outerPartitionId = 0;
-        int innerPartitionId = 1;
-        for (; innerPartitionId < hz.getPartitionService().getPartitions().size(); innerPartitionId++) {
-            if (!mappedToSameThread(operationService, outerPartitionId, innerPartitionId)) {
-                break;
-            }
-        }
-
-        InnerOperation innerOperation = new InnerOperation(response, innerPartitionId);
-        OuterOperation outerOperation = new OuterOperation(innerOperation, outerPartitionId);
+    public void invokeOnPartition_outerLocal_innerSameInstance_callsDifferentPartition() {
+        HazelcastInstance local = createHazelcastInstance();
+        OperationService operationService = getOperationService(local);
 
+        int outerPartitionId = getPartitionId(local);
+        int innerPartitionId = randomPartitionIdNotMappedToSameThreadAsGivenPartitionIdOnInstance(local, outerPartitionId);
+        InnerOperation innerOperation = new InnerOperation(RESPONSE, innerPartitionId);
+        OuterOperation outerOperation = new OuterOperation(innerOperation, outerPartitionId);
         InternalCompletableFuture f = operationService.invokeOnPartition(null, outerOperation, outerOperation.getPartitionId());
 
-        try {
-            f.getSafely();
-            fail();
-        } catch (IllegalThreadStateException e) {
-        }
+        expected.expect(IllegalThreadStateException.class);
+        expected.expectMessage("cannot make remote call");
+        f.getSafely();
     }
 
     @Test
-    public void whenPartition_callsDifferentPartition_butMappedToSameThread() throws ExecutionException, InterruptedException {
+    public void invokeOnPartition_outerLocal_innerSameInstance_callsDifferentPartition_mappedToSameThread() throws ExecutionException, InterruptedException {
```

Comment:
There's already `InvocationNestedLocalTest` and `InvocationNestedRemoteTest`
Wouldn't like to split a test to two files because of a name that's descriptive and tells me everything about it. The name can be shorter actually since `callsDifferentPartition` is redundant.